### ISSUE TITLE
Consul API client

### ISSUE THREAD

**Issue by zigster64** [zigforum/needed-libraries#5]
https://www.consul.io/api-docs/libraries-and-sdks

I tend to end up using consul for quite a few things.   Its a Go project, so the Go API is well supported, but even the C/C++ support is pretty poor.

Its basically just a well behaved REST  client that needs to be written.

Blocker at the moment is really the evented IO issues that are pending to be fixed in Zig soonish.  Makes it hard to do robust RESTy type things till that net library is sorted.

See https://github.com/Luukdegram/apple_pie/issues/23. for context and link to the zig issuehttps://github.com/ziglang/zig/issues/8224

TL;DR - once the zig network evented IO fixes are released, I expect that floodgates will open to porting a lot of Webby things to Zig without much pain.

Naive (and possibly worthless) benchmarks of Zig vs popular "fast" environments like Go or Node or Crystal show that Zig already has much better load throughput, so its going to be good when its all fixed :)
